students/srepking/Lesson06/mailroom4.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+def more_choices():
+    while True:
+        name = input('\nChoose an Option: \n'
+                     'e - to exit\n'
+                     'list -To see a list of names, or\n'
+                     'Type a name to start your thank you letter >>')
+        if name == 'e':
+            return
+        if name == 'list':
+            create_list(donors)
+        else:
+            print('\n''Ok, you want to write a letter for {}, '
+                  'lets see what we can do.'.format(name))
+
+            if donors.get(name) is None:
+                yes_no = input('The name you entered is not in the database.'
+                               'Would you like to add this name? y or n >>')
+                if yes_no == 'n':
+                    return
+
+            amount = input('\n''What is the donation amount? or \'e\' to exit >')
+            if amount == 'e':
+                return
+            else:
+                thank_you(donors, name, amount)
+
+
+
+def create_list(some_donors):
+    # This prints the list of donors
+    for x in some_donors:
+        print(x)
+
+
+def thank_you(dict_donors, some_name, some_amount):
+    """Add a donation to a donors records and print a report."""
+
+    if dict_donors.get(some_name) is None:
+        dict_donors[some_name] = []
+    try:
+        if int(some_amount) <= 0:
+            print('\nYou entered an invalid amount!!\n')
+            return
+    except ValueError:
+        print('\nYou entered an invalid amount!!\n')
+        return ValueError
+
+    dict_donors[some_name].append(float(some_amount))
+    print('Thank you so much for the generous gift of ${0:.2f}, {1}!'
+          .format(float(some_amount), some_name))
+    return dict_donors
```

Comment:
return value in this function looks to be a test only code, since it isn't actually used - best to avoid code like that.